oap-server/oal-rt/src/main/java/org/apache/skywalking/oal/rt/OALRuntime.java

```diff
@@ -523,6 +524,9 @@ private void writeGeneratedFile(CtClass metricsClass, String className, String t
                 if (!folder.exists()) {
                     folder.mkdirs();
                 }
+                if (StringUtils.equalsIgnoreCase("dispatcher", type)) {
+                    className = metricsClass.getSimpleName();
+                }
```

Comment:
This looks to be not an elegant solution, maybe we can just simplify this method's signature to `private void writeGeneratedFile(CtClass metricsClass, String type) throws OALCompileException` and use `metricsClass.getSimpleName()` to replace the original argument `className`?